### PR TITLE
Refactor to remove build warnings

### PR DESCRIPTION
**Pull request by smyrick**
### :pencil: Description
Refactor library and test code to remove some build and IDE warnings. This is a non-functional change.

### :link: Related Issues
N\A

### CODE REVIEW

**Comment by dariuszkuc** [comment]
> The changes for the RouteConfiguration are there to provide more internal documentation and suppress the warning we had when calling jackson to read the post body from a suspend function. That is no longer relevant if we move it to its own private function.

Warning was valid - reading body is a blocking operation. Moving this code around doesn't change this. Personally I am also unsure whether the new refactored version of `RouteConfiguration` is that much easier to follow (i.e. you got to jump through multiple methods to see what is happening).

File: graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/execution/ApolloSubscriptionProtocolHandler.kt [hunk 1]
```diff
@@ -144,14 +144,22 @@ class ApolloSubscriptionProtocolHandler(
...
         val onConnect = subscriptionHooks.onConnect(connectionParams, session, graphQLContext)
         sessionState.saveOnConnectHook(session, onConnect)
         val acknowledgeMessageFlux = Flux.just(acknowledgeMessage)
         val keepAliveFlux = getKeepAliveFlux(session)
         return acknowledgeMessageFlux.concatWith(keepAliveFlux)
     }
 
+    @Suppress("UNCHECKED_CAST")
+    private fun convertPayloadToMap(payload: Any?): Map<String, String> =
+        if (payload is Map<*, *>) {
```
Comment (dariuszkuc) [inline graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/execution/ApolloSubscriptionProtocolHandler.kt]: whats the point of extra `is` check? e.g. why not just keep it simply `payload as? Map<String, String> ?: emptyMap()` as it used to be (and in that case do we really need one liner method?)?

File: graphql-kotlin-spring-server/src/test/kotlin/com/expediagroup/graphql/spring/routes/SubscriptionRoutesConfigurationIT.kt [hunk 2]
```diff
@@ -130,6 +130,8 @@ class SubscriptionRoutesConfigurationIT(
             .verify()
     }
 
+    private fun convertValueToString(message: SubscriptionOperationMessage) = objectMapper.writeValueAsString(message)
```
Comment (dariuszkuc) [inline graphql-kotlin-spring-server/src/test/kotlin/com/expediagroup/graphql/spring/routes/SubscriptionRoutesConfigurationIT.kt]: unsure how useful this method is -> it is just a test class anyway....

Comment (smyrick) [reply graphql-kotlin-spring-server/src/test/kotlin/com/expediagroup/graphql/spring/routes/SubscriptionRoutesConfigurationIT.kt]: Will delete

File: graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/execution/ApolloSubscriptionProtocolHandler.kt (hunk 1 above)
Comment (smyrick) [reply graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/execution/ApolloSubscriptionProtocolHandler.kt]: Yea, will delete. This was added so I could verify what the warning in intellij was complaining about.

**Pull request merged**